build-and-push-image/action.yml

```
@@ -0,0 +1,101 @@
+name: Build and Push
+description: Build docker image and push to AWS ECR
+
+inputs:
+  push_to_ecr:
+    description: Determines whether the image should be pushed
+    required: false
+    default: "false"  # ToDo: Switch default after ECR repositories are created
+  ecr_repository:
+    description: ECR repository to push image in
+    required: true
+  dockerfile_path:
+    description: Relative path to Dockerfile. Defaults to `Dockerfile`
+    required: false
+    default: Dockerfile
+
+  aws_access_key_id:
+    description: AWS access key
+    required: true
+  aws_secret_access_key:
+    description: AWS secret access key
+    required: true
+  aws_region:
+    description: AWS region
+    required: false
+    default: eu-central-1
+
+  ssh_key:
+    description: SSH key
+    required: true
```

SSH key is optional. For instance moneymeets-backoffice is not using this.
```suggestion
    required: false
```